samples/test/verify.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import os
+import serial
+from time import time
+import re
+import json
+
+def wait_for_regex_in_line(ser, regex, timeout_s=20, log=True):
+    start_time = time()
+    while True:
+        line = ser.readline().decode('utf-8', errors='replace').replace("\r\n", "")
+        if line != "" and log:
+            print(line)
+        if time() - start_time > timeout_s:
+            raise RuntimeError('Timeout')
+        regex_search = re.search(regex, line)
+        if regex_search:
+            return regex_search
+
+def set_setting(ser, key, value):
+    ser.write('\r\n'.encode())
+    wait_for_regex_in_line(ser, 'uart:', log=False)
+    ser.write('settings set {} {}\r\n'.format(key, value).encode())
+    wait_for_regex_in_line(ser, 'saved', log=False)
+
+def set_credentials(ser):
+    with open('credentials.json', 'r') as f:
+        settings = json.load(f)
+
+    print('===== Setting credentials via CLI (logging disabled) ====')
+
+    # WiFi credentials are options (nRF9160dk does not need them)
+    if 'wifi/ssid' in settings:
+        set_setting(ser, 'wifi/ssid', settings['wifi/ssid'])
+
+    if 'wifi/psk' in settings:
+        set_setting(ser,'wifi/psk', settings['wifi/psk'])
+
+    set_setting(ser,'golioth/psk-id', settings['golioth/psk-id'])
+    set_setting(ser,'golioth/psk', settings['golioth/psk'])
```

Comment:
Just minor comments about possible improvements:
1. I'd prefer `yaml` - just looks nicer IMO,
2. Would create something among those lines in `yaml`:
```
settings:
    golioth/psk-id: mypskid
    golioth/psk: mypsk
    wifi/ssid: myssid
    wifi/psk: mywifipsk
```
and load it by iterating through all `settings` node, so that we don't need explicitly list that in Python code.

Reply:
The main reason I went with JSON is that it is built-in to Python. But YAML is not a big deal, just need `pip install pyyaml`.